src/Raven.Client/Documents/Linq/RavenQueryProviderProcessor.cs

```diff
@@ -1985,6 +2022,34 @@ private void VisitQueryableMethodCall(MethodCallExpression expression)
             }
         }
 
+        private bool IsFilterModeActive() => DocumentQuery switch
+        {
+            DocumentQuery<T> documentQuery => documentQuery.IsFilterActive,
+            AsyncDocumentQuery<T> asyncDocumentQuery => asyncDocumentQuery.IsFilterActive,
+            _ => throw new InvalidDataException(($"Unknown query type for Filter.")) 
```

Comment:
1. Better to throw `ArgumentException` or `NotSupportedException`. Invalid data is usually reserved for server side errors with data integrity or something like that
2. Please also include the actual type of `DocumentQuery` in the exception message - `DocumentQuery.GetType()`

Reply:
done